File: 정민선/프로그래머스/k진수에서소수개수구하기.py
import math

def solution(n, k):
    answer = 0
    
    def change(num, c):
        result = ''
        while num > 0:
            num, mod = divmod(num, c)
            result += str(mod)
        return result[::-1]
    
    result = change(n, k).split('0')
    
    def primenumber(x):
        if x < 2:
            return False
        
        for i in range(2, int(math.sqrt(x) + 1)):
            if (x % i) == 0:
                return False
            
        return True	
    
    for r in result:
        if r == '':
            continue
            
        if primenumber(int(r)):
            answer += 1
    
    return answer

File: 이진영/프로그래머스/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    const strs = n.toString(k).split(/0+/);
    
    return strs.filter(v => isPrime(+v)).length;
}

const isPrime = n => {
    if (n < 2) return false;
    
    for (let i = 2; i <= Math.sqrt(n); i += 2) {
        if (n % i === 0) return false;
    }
    
    return true;
}